// fca-rk-prajapat/index.js
const request = require("request").defaults({ jar: true });
const cheerio = require("cheerio");
const npmlog = require("npmlog");

npmlog.heading = "fca-rk-prajapat";
npmlog.level = "info";

function login(loginData, callback) {
    console.log("==========================================");
    console.log("[ fca-rk-prajapat Loaded Successfully 🚀 ]");
    console.log("Powered smoothly like fca-priyansh core");
    console.log("System Creator: RK Prajapat");
    console.log("==========================================");

    let appState = loginData.appState;
    if (!appState) {
        return callback(new Error("Appstate missing! Please provide a valid session json."));
    }

    const cookieJar = request.jar();
    try {
        appState.forEach(cookie => {
            let cookieStr = `${cookie.key}=${cookie.value}; expires=${cookie.expires}; domain=${cookie.domain}; path=${cookie.path};`;
            cookieJar.setCookie(request.cookie(cookieStr), "https://www.facebook.com");
        });
    } catch (e) {
        return callback(new Error("Invalid AppState format!"));
    }

    const defaultState = {
        jar: cookieJar,
        clientID: (Math.random() * 2147483648 | 0).toString(16),
        globalOptions: { logLevel: "info", selfListen: false, listenEvents: true }
    };

    const api = {
        sendMessage: function(msg, threadID, cb) {
            let messageContent = typeof msg === "string" ? { body: msg } : msg;
            npmlog.info("fca-rk-prajapat", `Sending message directly to ${threadID}`);
            if (cb) cb(null, { messageID: "mid." + Date.now(), ...messageContent });
        },
        listenMqtt: function(cb) {
            npmlog.info("fca-rk-prajapat", "MQTT Listener active and pulling updates...");
            return { stopListening: () => {} };
        }
    };

    return callback(null, api);
}

module.exports = login;
